### PR TITLE
APM setup in Agent section

### PR DESCRIPTION
**Pull request by l0k0ms**
#### APM related

* [x] Moves APM Agent setup section in the agent section
* [x] Creates docker sub-section  
* [x] Creates a docker APM sub-sub-section
* [x] Creates a Kubernetes APM sub-sub-section
* [x] Updates K8s default manifest to take into account APM setup
* [x] Creates a Language sub-section in the APM section that list all language instrumentation setup.

#### Side effect

* [x] Moves agent network page to FAQs in order to limit the complexity of the Agent nav 
* [x] Moves custom python page to FAQs in order to limit the complexity of the Agent nav 

* [x] Updates all links 
* [x] Updates all images
* [x] Moves custom python package doc to faq

### Motivation

After having discussed with APM PMs it has been decided that every part of the APM setup related to the agent should live in the Agent section of the documentation in order to enhance clarity and reduce complexity. 

### Preview link

* [Agent Page](https://docs-staging.datadoghq.com/gus/agent-apm/agent/)
* [New Agent APM section](https://docs-staging.datadoghq.com/gus/agent-apm/agent/apm/?tab=agent630)
* [K8s APM](https://docs-staging.datadoghq.com/gus/agent-apm/agent/kubernetes/apm)
* [New docker section](https://docs-staging.datadoghq.com/gus/agent-apm/agent/docker/)
* [Docker APM page](https://docs-staging.datadoghq.com/gus/agent-apm/agent/docker/apm/?tab=java)
* [New language section](https://docs-staging.datadoghq.com/gus/agent-apm/tracing/languages/)

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -1258,6 +1270,122 @@ languages:
...
+          url: "agent/apm/dotnet/"
+          weight: 1570
+          parent: customnav_tracingnav_setup
+        - identifier: customnav_tracingnav_php
+          name: "PHP"
+          url: "agent/apm/php/"
+          weight: 1580
+          parent: customnav_tracingnav_setup
+        - identifier: docker_ecs
+          name: "Tracing ECS application"
```
Comment (irabinovitch) [inline config.yaml]: Also I dont think "ECS application" is the right way to phrase this.  The application is being run in a a container orchestrated on ECS, but its not an application specific to ECS.

Per Amazon guidelines:
Amazon Elastic Container Service
or
Amazon ECS

File: config.yaml [hunk 2]
```diff
@@ -124,6 +124,34 @@ languages:
...
 
+        ## TRACING/APM - Setup basic
+        - identifier: customnav_tracingnav_setup
+          name: "APM"
+          url: "agent/apm/"
+          weight: 13
+          parent: menu_references_agent
+
+        - identifier: docker_ecs
+          name: "Tracing ECS application"
```
Comment (irabinovitch) [inline config.yaml]: Is ECS different enough from other container environments to warrant its own section?

Per Amazon guidelines:
Amazon Elastic Container Service
or 
Amazon ECS

File: content/agent/apm/ecs.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
+---
+title: Tracing ECS Applications
+kind: Documentation
+aliases:
+    - /tracing/ecs/
+further_reading:
+- link: "integrations/amazon_ecs/"
+  tag: "Documentation"
+  text: "AWS ECS setup"
```
Comment (irabinovitch) [inline content/agent/apm/ecs.md]: Service is Amazon ECS, not AWS ECS. Amazon is a stickler on this.

File: config.yaml (hunk 1 above)
Comment (l0k0ms) [reply config.yaml]: I just moved the content for now to remove the other PR, I think this content could go the Datadog - Amazon Elastic Container Service integration page directly after all.

File: content/tracing/languages/_index.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+---
+title: Language instrumentation
+kind: Documentation
+aliases:
+    - /tracing/setup
+---
+
```
Comment (cswatt) [inline content/tracing/languages/_index.md]: Can we give this page some context here? like
```suggestion
After you have [installed the Datadog Agent][1], [enabled trace collection][2], and [configured your environment[3], begin tracing by instrumenting your application.

[1]: /agent
[2]: /agent/apm/?tab=agent630#agent-configuration
[3]: /apm/?tab=agent630#primary-tags
```

File: content/tracing/advanced_usage/_index.md [hunk 5]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Advanced APM Setup
+title: Advanced Languages instrumentation
```
Comment (cswatt) [inline content/tracing/advanced_usage/_index.md]: Can we call this just "Advanced instrumentation"? Seems like there's a lot on this page that doesn't pertain to languages. Also, it's nice to have the page heading match the navigation link.

File: content/tracing/languages/java.md [hunk 6]
```diff
@@ -18,7 +18,7 @@ further_reading:
 
 ## Installation and Getting Started
 
-To begin tracing applications written in any language, first [install and configure the Datadog Agent][3] (see additional documentation for [tracing Docker applications](/tracing/setup/docker/)).
+To begin tracing applications written in any language, first [install and configure the Datadog Agent][3] (see additional documentation for [tracing Docker applications](//agent/docker/apm/)).
```
Comment (cswatt) [inline content/tracing/languages/java.md]: Why does it link to `//agent/docker/apm/` instead of `/agent/docker/apm/`?

Comment (l0k0ms) [reply content/tracing/languages/java.md]: wrong link :)

File: content/tracing/languages/_index.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+After you have [installed the Datadog Agent][1], [enabled trace collection][2], and [configured your environment[3], begin tracing by instrumenting your application.
```
Comment (palazzem) [inline content/tracing/languages/_index.md]: `[configured your environment[3]` -> `[configured your environment][3]`

**Pull request closed without merging**